pylib/zeus/zeus.py

```diff
@@ -52,6 +93,26 @@ def _Init(self):
       parser.set_defaults(func=self._GetHandler(cmd, 'run'))
 
     Flags.InitArgs()
+    if not all((Flags.ARGS.id, Flags.ARGS.root)):
+      if Flags.ARGS.task != ['...']:
+          tasks_dirs = map(
+            lambda t: os.path.abspath(t) if os.path.isdir(t)
+                      else os.path.dirname(os.path.abspath(t)),
+            Flags.ARGS.task
+          )
+          base_dir = os.path.commonpath(tasks_dirs)
+          try:
+            zeus_executable = find_zeus_executable(base_dir)
+          except FileNotFoundError:
+            zeus_executable = None
+      else:
+          zeus_executable = find_zeus_executable(os.getcwd())
+
+      if zeus_executable:
+        print(f'Found Zeus executable at: {zeus_executable}')
+        execute_and_replace(zeus_executable)
+      else:
+        Flags.PARSER.error('--id and --root are required')
```

Comment:
That's not technically true. The executable wasn't found. --id and --root are not required.

Reply:
like this?

it's very zenysis-specific hack so I didn't care it to be 100% correct, more like 100% working.

correct way would be to replace those bash scripts with some kind of config because now there's a risk for an infinite recursion and those bashes are ugly and copy paste too

but 1st zeus can't be fixed, only rewritten, as I said before, and it'd require synchronized change in zenysis repo which also would be very big so I decided to take this shortcut as it's not for production anyway

Reply:
Agreed - as long as we don't make it worse.